idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/DependencyGroup.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions.intellij;
+
+import com.intellij.codeInsight.completion.CompletionParameters;
+import com.intellij.psi.PsiElement;
+import com.palantir.gradle.versions.intellij.psi.VersionPropsTypes;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public abstract class DependencyGroup {
+
+    @Value.Parameter
+    public abstract List<String> parts();
+
+    public static DependencyGroup fromString(String string) {
+        return ImmutableDependencyGroup.of(Arrays.asList(string.split("\\.")));
+    }
+
+    public final String asUrlString() {
+        String url = String.join("/", parts());
+
+        // For some odd reason if you edit the first part of the group you get "IntellijIdeaRulezzz"
+        if (url.contains("IntellijIdeaRulezzz")) {
```

Comment:
It feels weird to have this check at this level. If a `CompletionContributor` can always add this, we probably want to add some code there to stop it rather than than here?

Reply:
I think this just appears if the set of completions is empty for some reason:

> [A default string that is inserted to the file before completion to guarantee that there'll always be some non-empty element there](https://github.com/JetBrains/intellij-community/blob/528b777e4a780350d7c8b0ccd1574e9102ccb71b/platform/core-api/src/com/intellij/codeInsight/completion/CompletionUtilCore.java#L9)

Reply:
Perhaps `groupFromParameters` should return an `Optional<DependencyGroup>`, rather than returning an empty string here that the caller has to remember to check.

Reply:
This was caused by me forgetting to check if the current element was the name or group, this meant when you edited the first group part in a completed line it would incorrectly trigger the name logic, Ive now fixed this so this check isnt needed anymore